lib/rucio/db/sqla/models.py

```diff
@@ -1263,6 +1265,15 @@ class LifetimeExceptions(BASE, ModelBase):
                    ForeignKeyConstraint(['account'], ['accounts.account'], name='LIFETIME_EXCEPT_ACCOUNT_FK'))
 
 
+class VO(BASE, ModelBase):
```

Comment:
Won't VO can be a account attribute too  ? maybe an table account vo map can be useful =

Reply:
Yes, accounts will be linked to a specific VO. This will be done by encoding the VO in the "account" field in the db. e.g. "jdoe@vo1"
Due to restrictions from the fact that some existing users have instances with very large tables that would be time consuming to rebuild (if the structure of some tables changed), the best option is to encode this in the PK.
(This is also true of scopes)

The issue with having a separate table to capture this is that we want to allow vos to work without competing over namespace, and since the name is the PK in account, this means including information from the VO to make it unique. Think of "John Smith" and "Jane Smith" in 2 different VOs, they should both be able to use "jsmith" as they won't see the other in the system.

Reply:
It won't be an account attribute, but a `vo` column in the accounts table. A vo map-account table is not really needed and it makes things a bit complicated since you could have tricky corner cases to e.g. separate data/rules/metadata etc from the different VOs

So if a person has access to multiple VOs, it will be an account in each of this VOs. The user can still use the same proxy/DN to login to both VOs, the VO just has to be named when authenticating.

Reply:
If a person is part of several VOS, the same account / person will be in several entries in the account table, e.g., jdoe@vo1, jdoe@vo2, etc. Right?
About encoding the vo identifier in the PK, it breaks the relational model and won't enforce any contraints like foreign keys. Will it concern the scope column of the big tables like dids, replicas, etc ?